### ISSUE TITLE
Show login status indicator (and perhaps actual login UI) on puzz solver page

### ISSUE THREAD

**Issue by glasser** [robx/pzprjs#318]
The puzz.link puzzle solver has different functionality if you're logged in vs not (saving your "done" status) but there's no way to tell when solving if you are logged in.  I also find I get logged out pretty frequently, so it would be helpful to be able to tell if I'm logged in on the solver page.

As an extra bonus, an actual login box there would be nice but not necessary.

**Comment by robx** [comment]
I agree, I'd vaguely planned to do something like this. It could probably be integrated rather unobtrusively in the menu bar (i.e., a top level menu entry that would say something like "login" when you're logged out and offer a login form, and show your username when logged in, and clicking that would open a menu allowing to log out and potentially interact with the database (favorite, see previous times, ...)).
